### ISSUE TITLE
Query matches in mode name of entry is highlighted, but not matched, in combi mode.

### ISSUE THREAD

**Issue by HactarCE** [DaveDavenport/rofi#862]
## Version

`1.5.1-dirty (tags/1.5.1)`

## Configuration

[Output of `rofi -help`](https://gist.github.com/HactarCE/7be5cc7f0f231837f1960242a2129ad5)

## Launch Command

`rofi -show combi`

## Steps to reproduce

* Enable `run` mode as one of `combi-modi`
* Run `rofi -show combi`
* Type `run` into the prompt

* Each entry contains the mode name (in this example, `run`) and the name of the program

## What behaviour you see

* Only entries where _the name of the program_ contains `run` are selected, and the word `run` is highlighted in both the name of the program and mode name

## What behaviour you expect to see

One of the following:

A:
* Only entries where _the name of the program_ contains `run` are selected, and the word `run` is hightlighted in _only_ the name of the program

B:
* Entries where _either_ the name of the program _or_ the mode name contains `run` are selected, and the word `run` is highlighted in both the name of the program and the mode name

Even better would be a configuration option to switch between the two behaviors.

**Comment by DaveDavenport** [comment]
Yeah, the highlight is independent of the filtering (for speed issues).
The highlight is on the visible text (not on searched fields).  I hope to improve this in the future.

**Comment by luxferresum** [comment]
Is there any way to get behavior B?

**Comment by DaveDavenport** [comment]
It is not going to be trivial (at all) to fix the highlighting of the text in the current design, I don't expect this be changed soon. (beside possibly removing highlighting as it seems to cause confusion because of this reason.)

**Comment by luxferresum** [comment]
@DaveDavenport I would like to have it the other way around. I would like to keep the highlighting but change the *matching*. So I want to be able to enter `run firefox` to start firefox (and not match a window)

**Comment by DaveDavenport** [comment]
This is supported, see manpage.  `!r firefox`

**Comment by mightyiam** [comment]
Problem with `!r` is that `firefox !r` doesn't work. Ideally, I would type `firefox` and then narrow it down to the `run` or `drun` result by expanding it to `firefox run` or even the `firefox r` would suffice.
